Grid2DLSTM.lua

```diff
@@ -147,17 +147,24 @@ function Grid2DLSTM:buildModel()
 end
 
 function Grid2DLSTM:updateOutput(input)
-  if (self.step == 1) and not self.cells[0] then
-    -- the initial state of the cell/hidden states
-    print("Initializing the cell/hidden states")
-    self.cells = {[0] = {}}
 
-    for L=1,self.nb_layers do
-      local h_init = torch.zeros(input:size(1), self.outputSize):cuda()
-      table.insert(self.cells[0], h_init:clone())
-      table.insert(self.cells[0], h_init:clone()) -- extra initial state for prev_c
+  if self.step == 1 then
+    -- the initial state of the cell/hidden states
+    if self.userPrevCell then
+      -- print("Initializing the cell/hidden states with user previous cell")
+      self.cells = { [0] = self.userPrevCell }
+    else
+      -- print("Initializing the cell/hidden states with zero")
+      self.cells = {[0] = {}}
+
+      for L=1,self.nb_layers do
+        local h_init = torch.zeros(input:size(1), self.outputSize):cuda()
```

Comment:
You will have to fix this option so it can work without a gpu, i'm getting errors on cpu version

Reply:
I tried removing cuda() references but could not get it to work on the rnn-sin demo with 4x2 tensorr